### PR TITLE
Add OS arch to plugin provided dependencies

### PR DESCRIPTION
**Pull request by AlexeyTsvetkov**

### CODE REVIEW

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt [hunk 1]
```diff
@@ -61,20 +62,29 @@ class ComposePlugin : Plugin<Project> {
...
+            replaceWith = ReplaceWith("linux_x64")
+        )
+        val linux = linux_x64
+        @Deprecated(
+            "compose.desktop.windows is deprecated, use compose.desktop.windows_x64 instead",
+            replaceWith = ReplaceWith("windows_x64")
+        )
+        val windows = windows_x64
+        @Deprecated(
+            "compose.desktop.macos is deprecated, use compose.desktop.macosX64 instead",
```
Comment (olonho) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt:81]: `macos_x64`

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/osUtils.kt [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+}
+
+internal val currentTarget by lazy {
+    Target(currentOS, currentArch)
+}
+
+internal val currentArch by lazy {
+    val osArch = System.getProperty("os.arch")
+    when (osArch) {
+        "x86_64" -> Arch.X64
```
Comment (olonho) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/osUtils.kt:26]: Some JVMs have `amd64` here on x86.

Comment (olonho) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/osUtils.kt:26]: See https://github.com/JetBrains/skiko/blob/cad5b3df74791cc431ce500b555fe57d80d18963/skiko/buildSrc/src/main/kotlin/properties.kt#L32

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt (hunk 1 above)
Comment (AlexeyTsvetkov) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt:81]: fixed

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/osUtils.kt (hunk 2 above)
Comment (AlexeyTsvetkov) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/osUtils.kt:26]: fixed

**Review by olonho — APPROVED** [review]

**Pull request merged**